mbed_devices/mbed_tools/cli.py

```diff
@@ -29,6 +30,10 @@ def list_connected_devices(format):
 cli = list_connected_devices
 
 
+def _sort_devices_by_name(devices: Iterable[Device]) -> Iterable[Device]:
+    return sorted(devices, key=attrgetter("mbed_target.board_name"))
```

Comment:
Can `board_name` ever be `None`? I have a feeling sorted wont like it if there is a None thrown in with a load of strings.

Reply:
Definitely something I didn't consider. However, hopefully not: https://github.com/ARMmbed/mbed-targets/blob/master/mbed_targets/mbed_targets.py#L88 .

Reply:
Current JSON returned from the os site does not seem to contain `null` values either, so I feel relatively safe. Let's say I trust the interface. 😅

Reply:
Ah that looks like we are pretty safe 👍

Reply:
It might be more efficient to cast this `iterable` to a `list` a use `list.sort` as it sorts in place w/o copying. Maybe it's not important.

Reply:
Ran some tests and I don't think it's important.
```
from mbed_devices import Device
from mbed_targets import MbedTarget
from operator import attrgetter
import timeit


def build_devices():
    devices = []
    for i in range(100):
        device = Device(
            mbed_target=MbedTarget({"attributes": {"name": f"target_{i:0=3}"}}),
            serial_number=str(i),
            serial_port=None,
            mount_points=[],
        )
        devices.append(device)
    return devices


def test_sort():
    list(build_devices()).sort(key=attrgetter("mbed_target.board_name"))


def test_sorted():
    sorted(build_devices(), key=attrgetter("mbed_target.board_name"))


print("Sort:")
print(timeit.timeit(test_sort, number=1000))
print("Sorted:")
print(timeit.timeit(test_sorted, number=1000))
```

Hopefully above doesn't have side effects I didn't consider.

```
Sort:
0.22790048900000004
Sorted:
0.228326346
```

Results are similar. Now, `sort` will be more memory efficient, but I doubt this will be used to list thousands of devices. ;)